src/Routing/Router.php

```diff
@@ -58,4 +58,25 @@ private function generateI18n($name, $parameters = [], $referenceType = self::AB
 
         return parent::generate($name.$this->routeNameSuffix, $parameters, $referenceType);
     }
+
+    /**
+     * @todo use custom matcher?
+     * @todo handle matchRequest()
+     */
+    public function match($pathinfo)
+    {
+        $match = parent::match($pathinfo);
+
+        // @todo optimize?
+        if (false !== strpos($match['_route'], $this->routeNameSuffix)) {
```

Comment:
It should recognize suffix only if string ends with `$this->routeNameSuffix`, this will be true for `route_i18n_name` as well. We probably need to take string length into account.